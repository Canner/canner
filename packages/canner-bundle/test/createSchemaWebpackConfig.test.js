/* eslint import/no-unresolved: 0 */

import path from 'path';
import {
  createSchemaConfig,
} from '../src/utils/createWebpackConfig';

const replacePath = path.resolve(__dirname, '../../..');
expect.addSnapshotSerializer({
  test: val => typeof val === 'string' && val.indexOf(replacePath) !== -1,
  print: val => val.replace(replacePath, ''),
});
describe('createSchemaConfig', () => {
  const config = createSchemaConfig({
    schemaPath: 'schemaPath',
    schemaOutputPath: 'schemaOutputPath',
    resolveModules: 'resolveModules',
    resolveLoaderModules: 'resolveLoaderModules',
    tsConfigFile: 'tsConfigFile',
    plugins: ['plugins'],
    watch: true,
    devMode: true,
  });
  it('default schema config', () => {
    expect(config).toMatchSnapshot();
  });
});
