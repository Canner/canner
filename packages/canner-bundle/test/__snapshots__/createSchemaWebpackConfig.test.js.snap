// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSchemaConfig default schema config 1`] = `
Object {
  "entry": "schemaPath",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "include": /node_modules/,
        "test": /\\\\\\.mjs\\$/,
        "type": "javascript/auto",
      },
      Object {
        "oneOf": Array [
          Object {
            "test": /canner\\\\\\.schema\\\\\\.tsx\\?\\$/,
            "use": Array [
              Object {
                "loader": "canner-schema-loader",
              },
              Object {
                "loader": "ts-loader",
                "options": Object {
                  "compilerOptions": Object {
                    "jsx": "react",
                    "jsxFactory": "CannerScript",
                  },
                  "configFile": "tsConfigFile",
                  "transpileOnly": true,
                },
              },
            ],
          },
          Object {
            "loader": "ts-loader",
            "options": Object {
              "compilerOptions": Object {
                "module": "es2015",
              },
              "configFile": "tsConfigFile",
              "getCustomTransformers": [Function],
              "transpileOnly": true,
            },
            "test": /\\\\\\.tsx\\?\\$/,
          },
        ],
      },
      Object {
        "oneOf": Array [
          Object {
            "test": /\\(\\\\\\.schema\\\\\\.js\\|canner\\\\\\.def\\\\\\.js\\)\\$/,
            "use": Array [
              Object {
                "loader": "canner-schema-loader",
              },
              Object {
                "loader": "babel-loader",
                "options": Object {
                  "babelrc": false,
                  "plugins": Array [
                    Array [
                      Object {
                        "default": [Function],
                      },
                      Object {
                        "legacy": true,
                      },
                    ],
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "default": [Function],
                    },
                    Array [
                      Object {
                        "default": [Function],
                      },
                      Object {
                        "loose": false,
                      },
                    ],
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "default": [Function],
                    },
                    Array [
                      Object {
                        "default": [Function],
                      },
                      Object {
                        "libraryName": "antd",
                        "style": true,
                      },
                    ],
                  ],
                  "presets": Array [
                    Object {
                      "default": [Function],
                      "isPluginRequired": [Function],
                      "transformIncludesAndExcludes": [Function],
                    },
                    Array [
                      Object {
                        "default": [Function],
                      },
                      Object {
                        "pragma": "CannerScript",
                        "pragmaFrag": "CannerScript.Default",
                        "throwIfNamespace": false,
                      },
                    ],
                    Object {
                      "default": [Function],
                    },
                  ],
                },
              },
            ],
          },
          Object {
            "exclude": /node_modules/,
            "test": /\\\\\\.js\\$/,
            "use": Object {
              "loader": "babel-loader",
              "options": Object {
                "babelrc": false,
                "plugins": Array [
                  Array [
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "legacy": true,
                    },
                  ],
                  Object {
                    "default": [Function],
                  },
                  Object {
                    "default": [Function],
                  },
                  Object {
                    "default": [Function],
                  },
                  Object {
                    "default": [Function],
                  },
                  Object {
                    "default": [Function],
                  },
                  Object {
                    "default": [Function],
                  },
                  Object {
                    "default": [Function],
                  },
                  Array [
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "loose": false,
                    },
                  ],
                  Object {
                    "default": [Function],
                  },
                  Object {
                    "default": [Function],
                  },
                  Array [
                    Object {
                      "default": [Function],
                    },
                    Object {
                      "libraryName": "antd",
                      "style": true,
                    },
                  ],
                ],
                "presets": Array [
                  Object {
                    "default": [Function],
                    "isPluginRequired": [Function],
                    "transformIncludesAndExcludes": [Function],
                  },
                  Object {
                    "default": [Function],
                  },
                  Object {
                    "default": [Function],
                  },
                ],
              },
            },
          },
        ],
      },
      Object {
        "loader": "ignore-loader",
        "test": /\\\\\\.\\(le\\|c\\)ss\\$/,
      },
    ],
  },
  "output": Object {
    "filename": "schema.node.js",
    "libraryTarget": "commonjs",
    "path": "schemaOutputPath",
  },
  "plugins": Array [
    LimitChunkCountPlugin {
      "options": Object {
        "maxChunks": 1,
      },
    },
    NormalModuleReplacementPlugin {
      "newResource": /packages/canner-bundle/src/utils/mock,
      "resourceRegExp": /firebase/,
    },
    WatchIgnorePlugin {
      "paths": Array [
        /node_modules/,
      ],
    },
    "plugins",
  ],
  "resolve": Object {
    "extensions": Array [
      ".jsx",
      ".js",
      ".ts",
      ".tsx",
      "mjs",
    ],
    "modules": "resolveModules",
  },
  "resolveLoader": Object {
    "modules": "resolveLoaderModules",
  },
  "target": "node",
  "watch": true,
}
`;
